scripts/manila-share-sync.py

```diff
@@ -231,9 +232,15 @@ def sync_share_state(self, shares, dry_run=True):
                 if share_status == 'error_deleting':
                     self.share_force_delete(share_id)
                     continue
-                self.share_reset_state(share_id, 'error')
-                if share_status == 'deleting':
-                    self.share_delete(share_id)
+
+                if share_status == 'extending':
+                    self.share_reset_state(share_id, 'extending_error')
+                elif share_status == 'shrinking':
+                    self.share_reset_state(share_id, 'shrinking_error')
+                elif share_status in ['creating', 'deleting']:
+                    self.share_reset_state(share_id, 'error')
+                    if share_status == 'deleting':
+                        self.share_delete(share_id)
             else:
```

Comment:
similarly on Line 280, for shares with replicas, the active replica could be in 'extending' or 'shrinking' state as well?